### PR TITLE
Fix Cassandra read bug when user query has no where clause (fixes #24829)

### LINKED ISSUE #24829: [Bug]: Cassandra syntax error when user passes query with no where clause
### What happened?

The current method for query input parsing on CassandraIO assumes such query necessarily comes with a 'where' clause, which leads to an invalid resulting statement when this is not the case.

Example of the current behavior:

Input query:
```
select partition_col, other_col from keyspace.table
```

Resulting invalid query: 
```
select partition_col, other_col from keyspace.table AND token(partition_col) >= ? and token(partition_col) < ?
```

Resulting error:
```
SyntaxException: [...] mismatched input 'AND' expecting EOF (..., other_col from keyspace.table [AND]...)
```

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by lpmarques**
It would be very useful to be able to pass a user-defined query to CassandraIO for purposes of column selection only.

However, the current method for query parsing assumes such query necessarily comes with a 'where' clause, which leads to an invalid resulting statement when this is not the case.

This behavior has been reported on issue #24829.

The hereby proposed change fixes #24829 by checking if the user given query contains a where clause, appending a "WHERE" to it when it doesn't, instead of a "AND".

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lpmarques** [comment]
fixes #24829

**Comment by Abacn** [comment]
Thanks for the catch, would you mind adding a unit test for this (query has no where clause)?

**Comment by lpmarques** [comment]
@Abacn thank you for the quick feedback. A unit test has been added now.

File: sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java [hunk 1]
```diff
@@ -485,6 +489,39 @@ public void testReadWithQuery() throws Exception {
...
+                .withHosts(Collections.singletonList(CASSANDRA_HOST))
+                .withPort(cassandraPort)
+                .withKeyspace(CASSANDRA_KEYSPACE)
+                .withTable(CASSANDRA_TABLE)
+                .withMinNumberOfSplits(20)
+                .withQuery(query)
+                .withCoder(SerializableCoder.of(Scientist.class))
+                .withEntity(Scientist.class));
+
+    PAssert.thatSingleton(output.apply("Count", Count.globally())).isEqualTo(22L);
```
Comment (Abacn) [inline sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java:511]:
```suggestion
    PAssert.thatSingleton(output.apply("Count", Count.globally())).isEqualTo(NUM_ROWS);
```

File: CHANGES.md [hunk 2]
```diff
@@ -80,7 +80,7 @@
 
 ## Bugfixes
 
-* Fixed X (Java/Python) ([#X](https://github.com/apache/beam/issues/X)).
+* Avoids Cassandra syntax error when user-defined query has no where clause in it (Java/Python) ([#24829](https://github.com/apache/beam/issues/24829)).
```
Comment (Abacn) [inline CHANGES.md:83]:
```suggestion
* Avoids Cassandra syntax error when user-defined query has no where clause in it (Java) ([#24829](https://github.com/apache/beam/issues/24829)).
```

**Comment by lpmarques** [comment]
I've just noticed there are currently two unreleased versions on the changelog: 2.44.0 and 2.45.0. Should I disconsider the latter and move my change to the former one?

**Review by Abacn — APPROVED** [review]
LGTM, thanks!

**Comment by Abacn** [comment]
> I've just noticed there are currently two unreleased versions on the changelog: 2.44.0 and 2.45.0. Should I disconsider the latter and move my change to the former one?

It will be 2.45.0. 2.44.0 branch is already cut: https://github.com/apache/beam/tree/release-2.44.0

**Comment by lpmarques** [comment]
> LGTM, thanks!

Thanks for the fix up suggestions!

**Comment by lpmarques** [comment]
Hey, @Abacn. How you doing?

Do we have any forecast for when 2.45.0 (or at least 2.44.0) is going to be realeased?

Thanks in advance

**Comment by Abacn** [comment]
Hi @lpmarques should be in mid Feb. Beam releases every 6 weeks. 2.45.0 branch will be cut next week and usually it takes 4-6 weeks to release.

**Pull request merged**